### PR TITLE
Rename SelectStatementContext#isSameGroupByAndOrderByItems to isGroupByStartsWithOrderByItems(#25414)

### PR DESCRIPTION
**Pull request by yangsen00**
Rename SelectStatementContext#isSameGroupByAndOrderByItems to isGroupByStartsWithOrderByItems(#25414 )

### CODE REVIEW

File: infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java [hunk 1]
```diff
@@ -294,8 +296,20 @@ private String getOrderItemText(final TextOrderByItemSegment orderByItemSegment)
      *
      * @return group by and order by sequence is same or not
      */
-    public boolean isSameGroupByAndOrderByItems() {
-        return !groupByContext.getItems().isEmpty() && groupByContext.getItems().equals(orderByContext.getItems());
+    public boolean isGroupByStartsWithOrderByItems() {
```
Comment (strongduanmu) [inline infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java:299]: Hi @yangsen00, can you add e2e test for this logic change?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java (hunk 1 above)
Comment (yangsen00) [reply infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java:299]: Of course.but Can you give a Can you give an example？

Comment (TeslaCN) [reply infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java:299]: @yangsen00 You may refer to https://github.com/apache/shardingsphere/tree/618548ed8bfc07771bc659ff82b68fe6e9a5d5fd/test/e2e/sql/src/test/resources/cases/dql

**Comment by terrymanu** [comment]
Close because of staled.

**Pull request closed without merging**